### PR TITLE
feat: promisify clean

### PR DESCRIPTION
**Pull request by imatlopez**
##### Checklist

- [x] `npm install && npm test` passes
- [ ] tests are included 
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

Promisifying the clean action such that it is similar to #2220 and #2225

### CODE REVIEW

File: lib/clean.js [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
+  const buildDir = 'build'
 
   log.verbose('clean', 'removing "%s" directory', buildDir)
-  rm(buildDir, callback)
+  return util.promisify(rm)(buildDir)
 }
 
-module.exports = clean
+module.exports = function (gyp, argv, callback) {
+  clean(gyp, argv).then(callback.bind(undefined, null), callback)
```
Comment (DeeDeeG) [inline lib/clean.js:16]: Hi, this PR looks good to me other than a small thing!

There is a gray line `undefined` output to the console with this PR.

Example output:

Before:

```
% /Users/[me]/node-gyp/bin/node-gyp.js clean
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@10.24.1 | darwin | x64
gyp info ok 
```

After:

```
% /Users/[me]/node-gyp/bin/node-gyp.js clean
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@10.24.1 | darwin | x64
undefined
gyp info ok 
```

I think this line I am commenting on is implicitly returning a variable that exists but has value `undefined` to the callback. Whereas before the callback was simply called with no argument.

It's the difference between `console.log()` and `var a; console.log(a)`. One logs nothing, the other logs `undefined`.

My suggestion is to add parentheses to indicate calling the callback with no argument whatsoever if the Promise succeeded/was fulfilled. See code suggestion here:

```suggestion
  clean(gyp, argv).then(callback.bind(undefined, null)(), callback)
```

Comment (DeeDeeG) [reply lib/clean.js:16]: P.S. I admit I don't understand the need to `bind` the callback, and the output seems the same to me (i.e. there is no output) with or without using `.bind()`.

If it's possible to omit the `.bind()` in the Promise succeeded/fulfilled case, then I would personally think that's better for making the code simpler and more readable.

Comment (imatlopez) [reply lib/clean.js]: thanks, made the onFulfilled function simply call callback no args. Goal of mine is to replace the callbacks with promises altogether as the backbone.

**Review by DeeDeeG — APPROVED** [review]
FWIW (though I'm not a maintainer here) I approve because:

- It's functionally equivalent
- It uses Promises and modern `const` instead of callbacks and `var`

File: lib/clean.js (hunk 1 above)
Comment (DeeDeeG) [reply lib/clean.js]: Thanks for the explanation!

**Pull request closed without merging**